agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -34,18 +36,37 @@ class AtumContext private[agent] (
   val atumPartitions: AtumPartitions,
   val agent: AtumAgent,
   private var measures: Set[Measure] = Set.empty
-) {
+) extends Logging {
 
   def currentMeasures: Set[Measure] = measures
 
   def subPartitionContext(subPartitions: AtumPartitions): AtumContext = {
     agent.getOrCreateAtumSubContext(atumPartitions ++ subPartitions)(this)
   }
 
+  private def validateMeasureApplicability(measure: Measure, df: DataFrame): Unit = {
```

Comment:
Would make this a method of `Measure`, so it can actually be customized based on each function.

Reply:
Good idea actually. Measure can have this one implemented in it, and individual ones can override if needed. Especially useful for Measures that don't operate on any column, such as `count`

Reply:
changed